cmd/k8s-operator/sts.go

```diff
@@ -274,30 +288,78 @@ func (a *tailscaleSTSReconciler) createOrGetSecret(ctx context.Context, logger *
 		if len(tags) == 0 {
 			tags = a.defaultTags
 		}
-		authKey, err := a.newAuthKey(ctx, tags)
+		authKey, err = a.newAuthKey(ctx, tags)
 		if err != nil {
-			return "", err
+			return "", "", err
 		}
-
+	}
+	if !declarativeContainerBootConfig && authKey != "" {
 		mak.Set(&secret.StringData, "authkey", authKey)
 	}
+	if declarativeContainerBootConfig {
+		conf := ipn.ConfigVAlpha{
+			Version:   "alpha0",
+			AcceptDNS: "false",
+			Locked:    "false",
+			Hostname:  &stsC.Hostname,
+		}
+		if stsC.Connector != nil {
```

Comment:
Currently the tailscaled configfile is only used for Connector proxies and Connector proxies are only deployed with the tailscaled configfile, but this will change in future.